### PR TITLE
fix(components/lookup): set input box height to match other field types

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2599488](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2599488)

### CODE REVIEW

File: apps/integration/src/app/integrations/field-heights/field-heights.component.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }
+}
+
+:host-context(.sky-theme-modern) {
+  form .sky-form-group ::ng-deep {
+    .sky-input-box .sky-control-help {
+      height: 0 !important;
+    }
+  }
+}
```
Comment (johnhwhite) [inline apps/integration/src/app/integrations/field-heights/field-heights.component.scss]: Without this change, the inline help affects the input box height.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1491/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1491/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1491/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1491/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1491/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1491/playground/)

File: libs/components/forms/src/lib/modules/input-box/input-box.component.scss [hunk 2]
```diff
@@ -287,6 +279,12 @@ sky-input-box {
         }
 
         .sky-control-help {
+          /*
+          The help button should be above the input element to ensure it is clickable.
+          Select inputs have a z-index of 3 so that clicking the label area still opens the select.
+          */
+          z-index: 4;
+          height: 0;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/input-box/input-box.component.scss]: While using `height: 0` works, its existence isn't self-explanatory (i.e. we'd need to put a comment above this line of code explaining why it's here). 

The extra height to the field is coming from the `3px` margin-top, correct? Could we instead remove the `height: 0` and apply `-3px` to the bottom margin to make the solution self-explanatory?
```
margin: 3px 0 -3px $sky-margin-inline-compact;
```

Comment (johnhwhite) [reply libs/components/forms/src/lib/modules/input-box/input-box.component.scss]: Updated.

File: apps/integration/src/app/integrations/field-heights/field-heights.component.spec.ts [hunk 3]
```diff
@@ -45,7 +45,8 @@ describe('FieldHeightsComponent', () => {
     ).toBeTruthy();
   });
 
-  it('should be accessible', async () => {
-    await expectAsync(fixture.nativeElement).toBeAccessible();
-  });
+  // todo: enable when token a11y fix merges
+  // it('should be accessible', async () => {
+  //   await expectAsync(fixture.nativeElement).toBeAccessible();
+  // });
```
Comment (Blackbaud-SteveBrush) [inline apps/integration/src/app/integrations/field-heights/field-heights.component.spec.ts]: I'll wait to approve this PR until we've reviewed the tokens PR.

File: apps/integration/src/app/integrations/field-heights/field-heights.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    await fixture.whenStable();
+    expect(
+      fixture.debugElement.query(
+        By.css('[data-sky-id="character-counter-error"] > sky-status-indicator')
+      )
+    ).toBeTruthy();
+  });
+
+  // todo: enable when token a11y fix merges
+  // it('should be accessible', async () => {
```
Comment (Blackbaud-SteveBrush) [inline apps/integration/src/app/integrations/field-heights/field-heights.component.spec.ts]: Should this block be uncommented now?

Comment (johnhwhite) [reply apps/integration/src/app/integrations/field-heights/field-heights.component.spec.ts]: Good catch. Updated.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/1531

**Pull request merged**